src/DynamoCore/Graph/Nodes/VariableInputNode.cs

```diff
@@ -13,11 +14,30 @@ namespace Dynamo.Graph.Nodes
     /// </summary>
     public abstract class VariableInputNode : NodeModel
     {
+        /// <summary>
+        /// Protected constructor for deserialization
+        /// </summary>
+        /// <param name="inPorts"></param>
+        /// <param name="outPorts"></param>
+        protected VariableInputNode(IEnumerable<PortModel> inPorts, IEnumerable<PortModel> outPorts) :base(inPorts,outPorts)
+        {
+            VariableInputController = new BasicVariableInputNodeController(this);
+        }
+
         protected VariableInputNode()
         {
             VariableInputController = new BasicVariableInputNodeController(this);
         }
 
+        /// <summary>
+        /// A flag used during serialization to indicated that
+        /// the node has a variable number of input ports.
+        /// </summary>
+        public bool VariableInputPorts
```

Comment:
API Change : The `VariableInputPorts` property is added to facilitate serialization.